File: bin/Data/AUI/Periodic/Sources/code_afontdescription.cpp
// UIFontdescription application source code
#include <Urho3D/AUI/ASlider.h>
#include <Urho3D/AUI/AFontDescription.h>
#include <Urho3D/AUI/ATextField.h>
#include "PeriodicApp.h"

void PeriodicApp::setup_afontdescription( AWidget *layout )
{
    PODVector<AWidget*> dest;
    layout->SearchWidgetClass( "TBButton", dest );
    for (unsigned ii = 0; ii < dest.Size(); ii++)
        SubscribeToEvent(dest[ii], E_WIDGETEVENT, URHO3D_HANDLER(PeriodicApp, HandleAfontdescriptionEvent ));

    ASlider *steps = static_cast<ASlider*>(layout->GetWidget("fontstep"));
    if(steps)
        SubscribeToEvent(steps, E_WIDGETEVENT, URHO3D_HANDLER(PeriodicApp, HandleAfontdescriptionEvent ));
}

void PeriodicApp::HandleAfontdescriptionEvent(StringHash eventType, VariantMap& eventData)
{
    using namespace WidgetEvent;
    AWidget* widget = static_cast<AWidget*>(eventData[P_TARGET].GetPtr());
    if ( widget == NULL ) return;
    if (eventData[P_TYPE] == UI_EVENT_TYPE_CLICK)
    {
        if (widget->GetId() == "uifontdescriptioncode" )
        {
            AppLog( "UIFontdescription support : " + widget->GetId() + " was pressed " );
            ViewCode ( "Sources/code_afontdescription.cpp", widget->GetParent() );
        }
        if (widget->GetId() ==  "uifontdescriptionlayout" )
        {
            AppLog( "UIFontdescription support : " + widget->GetId() + " was pressed ");
            ViewCode ( "Scenes/layout_afontdescription.ui.txt", widget->GetParent() );
        }
    }
    else if (eventData[P_TYPE] == UI_EVENT_TYPE_CHANGED )
    {
        if ( widget->GetId() ==  "fontstep" )
        {
            ASlider *uis = static_cast<ASlider*>(widget);
            if (uis)
            {
                ATextField *mytext = static_cast<ATextField*>(widget->FindWidget("changetext"));
                AFontDescription *myfont = new AFontDescription(context_);
                myfont->SetSize(int(uis->GetValue()));
                myfont->SetId("Vera");
                mytext->SetFontDescription (myfont);
                mytext->SetText ( "Size " + String (int (uis->GetValue())));
                AppLog( "UIFontdescription action : " + widget->GetId() + " step size changed to " + String (uis->GetValue()));
            }
        }
    }
}

File: bin/Data/AUI/Periodic/LuaScripts/code_afontdescription.lua
-- UIFontdescription application source code


function setup_uifontdescription(  layout )
    local dest = layout:SearchAWidgetClass( "TBButton" )
    for i, awid in ipairs(dest) do
        SubscribeToEvent(awid, "WidgetEvent", "HandleUifontdescriptionEvent")
	end
    local sld = layout:GetWidget ("fontstep")
    if sld ~= nil then
		local d1 = tolua.cast(sld, "AWidget")
        SubscribeToEvent(d1, "WidgetEvent", "HandleUifontdescriptionEvent" )
	end
end

function HandleUifontdescriptionEvent( eventType,  eventData)
    local widget = eventData["Target"]:GetPtr("AWidget")
    if widget == nil then
 		return
	end
    if eventData["Type"]:GetInt() == UI_EVENT_TYPE_CLICK then
        if widget:GetId() == "uifontdescriptioncode" then
            AppLog( "UIFontdescription support : " .. widget:GetId() .. " was pressed " )
            ViewCode ( "LuaScripts/code_afontdescription.lua", widget:GetParent())
        end
        if widget:GetId() ==  "uifontdescriptionlayout" then
            AppLog( "UIFontdescription support : " .. widget:GetId() .. " was pressed ")
            ViewCode ( "Scenes/layout_afontdescription.ui.txt", widget:GetParent())
        end
    elseif eventData["Type"]:GetInt() == UI_EVENT_TYPE_CHANGED then
        if widget:GetId() == "fontstep" then
 			local uis2 = tolua.cast( widget, "ASlider")
            local mytext = widget:FindWidget("changetext")
            local myfont = AFontDescription:new()
            myfont:SetSize( math.floor(uis2:GetValue())) -- convert float to int
            myfont:SetId("Vera")
            mytext:SetFontDescription (myfont)
            mytext:SetText ( "Size " .. math.floor(uis2:GetValue()))
            AppLog( "UIFontdescription action : " .. widget:GetId() .. " step size changed to " .. math.floor(uis2:GetValue()))
        end
    end
end

